src/skarab_fpga.py

```diff
@@ -285,6 +289,56 @@ def read(self, device_name, size, offset=0):
         # return the number of bytes requested
         return data[offset: offset + size]
 
+    def bulk_read(self, device_name, size, offset=0):
+        """
+        Return size_bytes of binary data with carriage-return escape-sequenced.
+        :param device_name: name of memory device from which to read
+        :param size: how many bytes to read
+        :param offset: start at this offset, offset in bytes
+        :return: binary data string
+        """
+
+        # map device name to address, if can't find, bail
+        if device_name in self.memory_devices.keys():
+            addr = self.memory_devices[device_name].address
+        elif type(device_name) == int and 0 <= device_name < 2 ** 32:
```

Comment:
@paulprozesky I tested for an int to prevent the case of the user attempting to pass a float. Unless this isn't necessary?